db/db_impl/db_impl.cc

```diff
@@ -511,6 +513,11 @@ Status DBImpl::CloseHelper() {
   }
   mutex_.Unlock();
 
+  // Below check is added as recovery_error_ is not checked and it causes crash
+  // in DBSSTTest.DBWithMaxSpaceAllowedWithBlobFiles when space limit is
+  // reached.
+  error_handler_.GetRecoveryError().PermitUncheckedError();
+
```

Comment:
Could you please also elaborate on this scenario?

Reply:
I was getting the error mentioned here.  https://github.com/facebook/rocksdb/pull/7715#issuecomment-742460190. After going through the comments, I added the PermitUncheckedError() here. But I am not sure if there is a better place to add the PermitUncheckedError.

Reply:
Took an admittedly brief look at the `ErrorHandler` code. I think the place that's missing a `PermitUncheckedError` call is in `RecoverFromRetryableBGIOError` when `recovery_error_` is reset to `Status::OK` (on line 653).

Reply:
... then again, I think that could only be triggered by using the other `SetBGError` overload that takes an `IOStatus`.

Reply:
Yesterday, I added the ```PermitUncheckedError``` at line https://github.com/facebook/rocksdb/blob/3dff28cf9b12d5bb2ac8bf339c2f635d57c77a1c/db/error_handler.cc#L242 and it ran succesfully so I added in the ```CloseHelper```.  I tried at Line 653 but that didn't work.